packages/medusa/src/services/product-variant.ts

```diff
@@ -245,86 +254,197 @@ class ProductVariantService extends TransactionBaseService {
+      ? ProductVariant[]
+      : ProductVariant
+  >(
+    variantOrVariantIdOrData: TInput,
+    updateData?: UpdateProductVariantInput
+  ): Promise<TResult> {
+    let data = Array.isArray(variantOrVariantIdOrData)
+      ? variantOrVariantIdOrData
+      : ([] as UpdateProductVariantData)
+
     return await this.atomicPhase_(async (manager: EntityManager) => {
       const variantRepo = manager.getCustomRepository(
         this.productVariantRepository_
       )
 
-      let variant = variantOrVariantId
-      if (typeof variant === `string`) {
-        const variantRes = await variantRepo.findOne({
-          where: { id: variantOrVariantId as string },
-        })
-        if (!isDefined(variantRes)) {
-          throw new MedusaError(
-            MedusaError.Types.NOT_FOUND,
-            `Variant with id ${variantOrVariantId} was not found`
+      if (updateData) {
+        let variant: Partial<ProductVariant> | null =
+          variantOrVariantIdOrData as Partial<ProductVariant>
+
+        if (isString(variantOrVariantIdOrData)) {
+          const variantRes = await variantRepo.findOne(
+            { id: variantOrVariantIdOrData },
+            {
+              select: variantRepo.metadata.columns.map(
+                (c) => c.propertyName
+              ) as (keyof ProductVariant)[],
+            }
           )
-        } else {
-          variant = variantRes as ProductVariant
+          if (!isDefined(variantRes)) {
+            throw new MedusaError(
+              MedusaError.Types.NOT_FOUND,
+              `Variant with id ${variantOrVariantIdOrData} was not found`
+            )
+          } else {
+            variant = variantRes
+          }
         }
-      } else if (!variant.id) {
-        throw new MedusaError(
-          MedusaError.Types.INVALID_DATA,
-          `Variant id missing`
-        )
-      }
 
-      const { prices, options, metadata, inventory_quantity, ...rest } = update
-
-      if (prices) {
-        await this.updateVariantPrices(variant.id!, prices)
-      }
-
-      if (options) {
-        for (const option of options) {
-          await this.updateOptionValue(
-            variant.id!,
-            option.option_id,
-            option.value
+        if (!variant?.id) {
+          throw new MedusaError(
+            MedusaError.Types.INVALID_DATA,
+            `Variant id missing`
           )
         }
-      }
 
-      if (typeof metadata === "object") {
-        variant.metadata = setMetadata(variant as ProductVariant, metadata)
+        data = [{ variant: variant as ProductVariant, updateData: updateData }]
       }
 
-      if (typeof inventory_quantity === "number") {
-        variant.inventory_quantity = inventory_quantity as number
-      }
+      const result = await this.updateBatch(data)
+
+      return (Array.isArray(variantOrVariantIdOrData)
+        ? result
+        : result[0]) as unknown as TResult
+    })
+  }
+
+  protected async updateBatch(
```

Comment:
nit: Should we rename to `updateBulk` to be consistent with our repositories?

Reply:
generally we are using `batch`, I ve only used `bulk` to distinguish the original insert from the custom one and to not call the custom one `insert` in order to not override the original implementation

Reply:
Got it. Fine to keep it as is :)